template/greeter/Dockerfile

```diff
@@ -2,10 +2,14 @@
 
 FROM --platform=${BUILDPLATFORM:-<%- platformName %>} node:18-alpine AS dev-deps
 
+RUN apk --no-cache add curl
+RUN corepack enable
+RUN corepack prepare pnpm@latest --activate
+RUN pnpm config set store-dir /workdir/.pnpm-store
+
 WORKDIR /workdir
 
 COPY package.json yarn.lock ./
 
-RUN \
-  --mount=type=secret,id=npm,dst=/root/.npmrc \
-  yarn install --frozen-lockfile --ignore-optional --non-interactive
+RUN --mount=type=secret,id=npm,dst=/root/.npmrc pnpm fetch
+RUN pnpm install --offline --recursive
```

Comment:
This is largely copied from some of my earlier experimentation; we should see if there are better implementations out there that we should copy instead.